### DISCUSSION TITLE
What do I do if I have multiple things in a PR?

### DISCUSSION THREAD

**Discussion by balloman** [semantic-release/semantic-release#3044] [Q&A] [answered]
I've encountered a situation, and I still haven't figured out how to proceed. I have a pull request. This pull request does multiple things (a feature and a fix). How do I then have a release created that has both the feature AND the fix in the changelog? Currently, only whichever is first in the commit message is recognized.

**Comment by travi** [comment, marked as answer]
semantic-release acts on commits, not on PRs. As long as you merge the PR so that the commits are preserved as individual commits, semantic-release will note both separately in the release notes and consider both in determining the proper version bump. 

tl;dr: don't squash you PR

**Reply by balloman** [reply]
Maybe this is out of scope of this package, but how could you attach multiple fixes to a single commit in that case? Not squashing is not an option.

**Reply by travi** [reply]
https://semantic-release.gitbook.io/semantic-release/support/troubleshooting#squashed-commits-are-ignored-by-semantic-release

yes, if you are choosing to use a workflow that requires always squashing, your workflow is outside the scope of semantic-release if you are also not making atomic changes
